hooks/hookexecution/executor.go

```diff
@@ -0,0 +1,187 @@
+	ExecuteAuctionResponseStage(response *openrtb2.BidResponse)
+}
+
+type HookStageExecutor interface {
+	StageExecutor
+	SetAccount(account *config.Account)
+	GetOutcomes() []StageOutcome
+}
+
+type hookExecutor struct {
+	account        *config.Account
+	accountId      string
+	endpoint       string
+	planBuilder    hooks.ExecutionPlanBuilder
+	stageOutcomes  []StageOutcome
+	moduleContexts *moduleContexts
+	// Mutex needed for BidderRequest and RawBidderResponse Stages as they are run in several goroutines
+	sync.Mutex
+}
+
+func NewHookExecutor(builder hooks.ExecutionPlanBuilder, endpoint string) *hookExecutor {
+	return &hookExecutor{
+		endpoint:       endpoint,
+		planBuilder:    builder,
+		stageOutcomes:  []StageOutcome{},
+		moduleContexts: &moduleContexts{ctxs: make(map[string]hookstage.ModuleContext)},
+	}
+}
+
+func (e *hookExecutor) SetAccount(account *config.Account) {
+	if account == nil {
+		return
+	}
+
+	e.account = account
+	e.accountId = account.ID
+}
+
+func (e *hookExecutor) GetOutcomes() []StageOutcome {
+	return e.stageOutcomes
+}
+
+func (e *hookExecutor) ExecuteEntrypointStage(req *http.Request, body []byte) ([]byte, *RejectError) {
+	plan := e.planBuilder.PlanForEntrypointStage(e.endpoint)
+	if len(plan) == 0 {
+		return body, nil
+	}
+
+	stageName := hooks.StageEntrypoint.String()
+	handler := func(
+		ctx context.Context,
+		moduleCtx hookstage.ModuleInvocationContext,
+		hook hookstage.Entrypoint,
+		payload hookstage.EntrypointPayload,
+	) (hookstage.HookResult[hookstage.EntrypointPayload], error) {
+		return hook.HandleEntrypointHook(ctx, moduleCtx, payload)
+	}
+
+	executionCtx := executionContext{
+		endpoint:       e.endpoint,
+		stage:          stageName,
+		accountId:      e.accountId,
+		account:        e.account,
+		moduleContexts: e.moduleContexts,
+	}
+
+	payload := hookstage.EntrypointPayload{Request: req, Body: body}
+	stageOutcome, payload, stageModuleContexts, rejectErr := executeStage(executionCtx, plan, payload, handler)
+	stageOutcome.Entity = entityHttpRequest
+	stageOutcome.Stage = stageName
+
+	e.saveModuleContexts(stageModuleContexts)
+	e.pushStageOutcome(stageOutcome)
+
+	return payload.Body, rejectErr
+}
+
+func (e *hookExecutor) ExecuteRawAuctionStage(body []byte) ([]byte, *RejectError) {
+	//TODO: implement
+	return nil, nil
+}
+
+func (e *hookExecutor) ExecuteProcessedAuctionStage(req *openrtb2.BidRequest) *RejectError {
+	//TODO: implement
+	return nil
+}
+
+func (e *hookExecutor) ExecuteBidderRequestStage(req *openrtb2.BidRequest, bidder string) *RejectError {
+	//TODO: implement
+	return nil
+}
+
+func (e *hookExecutor) ExecuteRawBidderResponseStage(response *adapters.BidderResponse, bidder string) *RejectError {
+	//TODO: implement
+	return nil
+}
+
+func (e *hookExecutor) ExecuteAllProcessedBidResponsesStage(responses []*adapters.BidderResponse) {
+	//TODO: implement
+}
+
+func (e *hookExecutor) ExecuteAuctionResponseStage(response *openrtb2.BidResponse) {
+	//TODO: implement
+}
```

Comment:
Is it necessary to have these functions defined if there is no code actually present in them yet?

Reply:
Yes, we can remove it from this PR and add it in subsequent PRs with implementation of these methods.